crates/scheduler/src/bin/hypha-scheduler.rs

```diff
@@ -491,22 +508,81 @@ async fn run(config: ConfigWithMetadata<Config>) -> Result<()> {
     Ok(())
 }
 
-fn get_executor_with_dataset(
+fn build_train_executor(
     config: &DiLoCo,
-    peer_id: PeerId,
-    dataset: String,
+    scheduler_peer: PeerId,
+    dataset: &str,
     parameter_server: PeerId,
     batch_size: u64,
+    epochs: u64,
 ) -> Executor {
-    Executor::DiLoCoTransformer {
-        model: config.model.clone().into(),
-        data: Fetch::scheduler(peer_id, dataset),
-        updates: Receive::peers(vec![parameter_server]),
-        results: Send::peers(vec![parameter_server], SelectionStrategy::One),
-        config: WorkerConfig::to_diloco(config.worker.clone(), batch_size as u32),
+    let descriptor = TrainExecutorDescriptor::new(TRAIN_EXECUTOR_NAME);
+    let model = build_model(&config.model);
+    let data = Fetch::scheduler(scheduler_peer, dataset.to_string());
+    let updates = Send::peers(vec![parameter_server], SelectionStrategy::All);
+    let results = Receive::peers(vec![parameter_server]);
+    let preprocessor = config
+        .preprocessor
+        .as_ref()
+        .map(|source| huggingface_fetch(source, HFRepoType::Model));
+
+    let train_config = TrainExecutorConfig {
+        model,
+        data,
+        updates,
+        results,
+        optimizer: config.inner_optimizer.clone(),
+        batch_size: i32::try_from(batch_size).unwrap_or(i32::MAX),
```

Comment:
I think `2147483647` is a bit large. Why not something like `2**13` which is allready huge?